src/modules/fundingManager/bondingCurveFundingManager/BondingCurveFundingManagerBase.sol

```diff
@@ -0,0 +1,143 @@
+import {ERC20Upgradeable} from "@oz-up/token/ERC20/ERC20Upgradeable.sol";
+
+// External Libraries
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+import {Strings} from "@oz/utils/Strings.sol";
+
+import {IFundingManager} from "src/modules/fundingManager/IFundingManager.sol";
+
+abstract contract BondingCurveFundingManagerBase is
+    IBondingCurveFundingManagerBase,
+    IFundingManager,
+    ContextUpgradeable,
+    ERC20Upgradeable,
+    Module
+{
+    using Strings for uint;
+    using SafeERC20 for IERC20;
+
+    //--------------------------------------------------------------------------
+    // Storage
+    uint8 internal tokenDecimals;
+    bool internal buyIsOpen;
+    uint internal buyFee;
+    address internal collateral;
+    // Q: Do we want to set other beneficiary in the contract then the module itself for the fees?
+
+    //--------------------------------------------------------------------------
+    // Public Functions
+
+    function token() public view returns (IERC20) {
+        return __Module_orchestrator.token();
+    }
+
+    function buyOrder(uint _depositAmount) external payable virtual {
+        // WiP
+        // Deduct fee from incoming value. Fee is paid in collateral token
+        _issueTokens(_depositAmount, collateral);
+    }
+
+    function openBuy() external onlyAuthorizedOrManager {
+        // Function to set the PAMM buy functionality to open
+        _openBuy();
+    }
+
+    function closeBuy() external onlyAuthorizedOrManager {
+        // Function to set the PAMM buy functionality to close
+        _closeBuy();
+    }
+
+    function updateBuyFee(uint _fee) external onlyAuthorizedOrManager {
+        // Should update the buy fee of the contract
+        _updateBuyFee(_fee);
+    }
+    //--------------------------------------------------------------------------
+    // Public Mutating Functions
+
+    function decimals()
+        public
+        view
+        override(ERC20Upgradeable)
+        returns (uint8)
+    {
+        return tokenDecimals;
+    }
+
+    function deposit(uint amount) external {}
+
+    function depositFor(address to, uint amount) external {}
+
+    function withdraw(uint amount) external {}
+
+    function withdrawTo(address to, uint amount) external {}
+
+    //--------------------------------------------------------------------------
+    // Internal Functions Implemented in Downstream Contract
+    function _issueTokensFormulaWrapper(
+        uint _depositAmount,
+        address _collateral
+    ) internal virtual returns (uint);
+
+    //--------------------------------------------------------------------------
+    // Internal Functions
+
+    function _openBuy() internal {
+        // Validate if not open
+        buyIsOpen = true;
+    }
+
+    function _closeBuy() internal {
+        // Validate if not closed
+        buyIsOpen = false;
+    }
+
+    function _setCollateral(address _collateral) internal {
+        collateral = _collateral;
```

Comment:
Isnt the collateral just the address(this).balance?
Wasnt that the main point of the Virtual Collateral to be able to be set?

Reply:
Could you explain this point again for me please. You are right that the collateral address is just the token of defined by the Orchestrator contract, However we talked about also the future possibility to add other collateral tokens in our first call, which is why I included this setter function.

I'm unsure what is meant by balance

Reply:
Forget what I said here 🙈
This is to set the underlying payment token address right?
Because I apparently cant read I assumed that this would somehow set a "virtual" Collateral Amount here, which it straight up doesnt